examples/cpp/customserializable/main.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+
+#include <geode/CacheFactory.hpp>
+#include <geode/PoolManager.hpp>
+#include <geode/PdxWrapper.hpp>
+
+using namespace apache::geode::client;
+
+class Order : public PdxSerializable {
+ public:
+  Order() : order_id_(0), name_(), quantity_(0) {};
+
+  Order(uint32_t order_id, std::string&& name, uint16_t quantity)
```

Comment:
By doing it this way you're making it illegal to pass name by lvalue.  Is that intentional?

Reply:
👍 good eye, I totally missed that they got rid of the l-value constructor.

Reply:
I did that because it wasn't being used in the example and to keep unused code to a minimum, but this shouldn't matter anyways after we switch to the universal reference constructor.

Reply:
I did that because it wasn't being used in the example and to keep unused code to a minimum, but this shouldn't matter anyways after we switch to the universal reference constructor.

Reply:
Perhaps this is a side topic, but how much C++11 knowledge do we expect of our users for barrier to entry?  Given that we are a Java first product I would think not much.  If that's the case then I would vote to use your original constructor even though it invokes an extra copy for a few reasons:

1) I think reference semantics is confusing for a novice C++11 developer and using them could detract from the core example we are trying to convey here.
2) If our user is a more advanced C++11 developer then they should already know how to do this themselves.
3) I think there is _sometimes_ a use case for invoking an extra copy.  Perhaps for readability and code that is not executed in a tight loop it may actually be the right way to go.

Reply:
Good points David. 👍

Reply:
Thanks!  Ended up going back with the original constructor for simplicity as suggested.